csrc/scheduler/utils.cpp

```diff
@@ -2848,9 +2848,29 @@ std::vector<int64_t> reorderDomainLike(
     }
   }
 
-  NVF_ERROR(std::ranges::all_of(permutation, [&](int64_t pos) {
-    return pos >= 0 && pos < (int64_t)permutation.size();
-  }));
+  // domain_to_reorder can be partial with respect to ref, that is,
+  // ordered_domain may contain IDs that do not appear in
+  // domain_to_reorder. In that case, at this point, the permutation
+  // vector may be sparse, e.g., {2, 0, 3}, which needs to be packed
+  // to {1, 0, 2}.
+  if (std::ranges::max(permutation) >= (int64_t)permutation.size()) {
+    auto permutation_copy = permutation;
+    std::ranges::sort(permutation_copy);
+    for (auto& pos : permutation) {
+      auto it = std::ranges::find(permutation_copy, pos);
+      NVF_ERROR(it != permutation_copy.end());
+      pos = static_cast<int64_t>(std::distance(permutation_copy.begin(), it));
+    }
+  }
+
+  NVF_ERROR(
+      std::ranges::all_of(
+          permutation,
+          [&](int64_t pos) {
+            return pos >= 0 && pos < (int64_t)permutation.size();
+          }),
```

Comment:
nitpick: might as well do a full permutation check here.
```suggestion
      std::ranges::is_permutation(
          permutation,
          std::ranges::iota(0, (int64_t) permutation.size())
          ),
```